vendor/github.com/OpenBazaar/spvwallet/peers.go

```diff
@@ -236,124 +209,75 @@ func (pm *PeerManager) onVerack(p *peer.Peer, msg *wire.MsgVerAck) {
 }
 
-func (pm *PeerManager) setDownloadPeer(peer *peer.Peer) {
-	log.Infof("Setting peer%d (%s) as download peer\n", peer.ID(), peer.UserAgent())
-	pm.downloadPeer = peer
-	if pm.startChainDownload != nil {
-		pm.blockQueue = make(chan chainhash.Hash, 32)
-		go pm.startChainDownload(pm.downloadPeer)
-	}
-}
-
-func (pm *PeerManager) QueueTxForDownload(peer *peer.Peer, txid chainhash.Hash, height int32) {
-	pm.peerMutex.Lock()
-	defer pm.peerMutex.Unlock()
-	queue, ok := pm.downloadQueues[peer.ID()]
-	if !ok {
-		queue = make(map[chainhash.Hash]int32)
-		pm.downloadQueues[peer.ID()] = queue
-	}
-	queue[txid] = height
-}
+// Called by connManager when it adds a new connection
+func (pm *PeerManager) getNewAddress() (net.Addr, error) {
+	// If we have a trusted peer we'll just return it
+	if pm.trustedPeer == nil {
+		pm.peerMutex.Lock()
+		defer pm.peerMutex.Unlock()
+		// We're going to loop here and pull addresses from the addrManager until we get one that we
+		// are not currently connect to or haven't recently tried.
+	loop:
+		for tries := 0; tries < 100; tries++ {
+			ka := pm.addrManager.GetAddress()
+			if ka == nil {
+				continue
+			}
 
-func (pm *PeerManager) DequeueTx(peer *peer.Peer, txid chainhash.Hash) (int32, error) {
-	pm.peerMutex.Lock()
-	defer pm.peerMutex.Unlock()
-	queue, ok := pm.downloadQueues[peer.ID()]
-	if !ok {
-		return 0, errors.New("Transaction not found")
-	}
-	height, ok := queue[txid]
-	if !ok {
-		return 0, errors.New("Transaction not found")
-	}
-	delete(queue, txid)
-	return height, nil
-}
+			// only allow recent nodes (10mins) after we failed 30
+			// times
+			if tries < 30 && time.Since(ka.LastAttempt()) < 10*time.Minute {
+				continue
+			}
 
-// Iterates over our peers and sees if any are reporting a height
-// greater than our height. If so switch them to the download peer
-// and start the chain download again.
-func (pm *PeerManager) CheckForMoreBlocks(height uint32) bool {
-	pm.peerMutex.RLock()
-	defer pm.peerMutex.RUnlock()
+			// allow nondefault ports after 50 failed tries.
+			if tries < 50 && fmt.Sprintf("%d", ka.NetAddress().Port) != pm.peerConfig.ChainParams.DefaultPort {
+				continue
+			}
 
-	moar := false
-	for peer := range pm.readyPeers {
-		if uint32(peer.LastBlock()) > height {
-			pm.downloadPeer = peer
-			go pm.startChainDownload(peer)
-			moar = true
-		}
-	}
-	return moar
-}
+			knownAddress := ka.NetAddress()
 
-// Called by connManager when it adds a new connection
-func (pm *PeerManager) getNewAddress() (net.Addr, error) {
-	if pm.trustedPeer == nil {
-		ka := pm.addrManager.GetAddress()
-		if ka == nil {
-			return &net.TCPAddr{}, errors.New("Adder manager returned nil address")
-		}
-		knownAddress := ka.NetAddress()
-		addr := &net.TCPAddr{
-			Port: int(knownAddress.Port),
-			IP:   knownAddress.IP,
+			// Don't return addresses we're still connected to
+			for _, p := range pm.connectedPeers {
+				if p.NA().IP.String() == knownAddress.IP.String() {
+					continue loop
```

Comment:
Why a labeled continue here and not extract this whole thing as a function call that returns an error you can check?

I tend to avoid labels as much as possible. An extracted function is more descriptive (added function name, clear input/return values), less opportunity for mutation errors should someone add behavior here later (like getting surprised with a continue that maintains state of an added variable versus as return which resets the value of the variable to a known state, zero, upon return), and much easier to reason about in your head instead of having to search around a largish function for the entry/exit points.